src/nut.c

```diff
@@ -123,6 +168,121 @@ static void nut_submit(nut_ups_t *ups, const char *type,
   plugin_dispatch_values(&vl);
 } /* void nut_submit */
 
+static int nut_connect(nut_ups_t *ups){
+#ifdef WITH_UPSCLIENT_27
```

Comment:
this function is getting overly long. Any chance you could split is out into something like nut_connect_tls?